src/debugger/direct/directDebugAdapter.ts

```diff
@@ -0,0 +1,198 @@
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as path from "path";
+import { ReactNativeProjectHelper } from "../../common/reactNativeProjectHelper";
+import { ErrorHelper } from "../../common/error/errorHelper";
+import { getExtensionVersion } from "../../common/extensionHelper";
+import { ILaunchArgs } from "../../extension/launchArgs";
+import { getProjectRoot } from "../nodeDebugWrapper";
+import { Telemetry } from "../../common/telemetry";
+import { OutputEvent, Logger } from "vscode-debugadapter";
+import { TelemetryHelper } from "../../common/telemetryHelper";
+import { RemoteTelemetryReporter } from "../../common/telemetryReporters";
+import { ChromeDebugAdapter, ChromeDebugSession, IChromeDebugSessionOpts, IAttachRequestArgs, logger } from "vscode-chrome-debug-core";
+import { InternalErrorCode } from "../../common/error/internalErrorCode";
+import { RemoteExtension } from "../../common/remoteExtension";
+import { DebugProtocol } from "vscode-debugprotocol";
+import { getLoggingDirectory } from "../../extension/log/LogHelper";
+import * as nls from "vscode-nls";
+import * as Q from "q";
+const localize = nls.loadMessageBundle();
+
+export interface IDirectAttachRequestArgs extends IAttachRequestArgs, ILaunchArgs {
+    cwd: string; /* Automatically set by VS Code to the currently opened folder */
+}
+
+export interface IDirectLaunchRequestArgs extends DebugProtocol.LaunchRequestArguments, IDirectAttachRequestArgs { }
+
+export class DirectDebugAdapter extends ChromeDebugAdapter {
+
+    private outputLogger: (message: string, error?: boolean | string) => void;
+    private projectRootPath: string;
+    private remoteExtension: RemoteExtension;
+    private isSettingsInitialized: boolean; // used to prevent parameters' reinstallation when attach is called from launch
+    private previousAttachArgs: IDirectAttachRequestArgs;
+
+    public constructor(opts: IChromeDebugSessionOpts, debugSession: ChromeDebugSession) {
+        super(opts, debugSession);
+        this.outputLogger = (message: string, error?: boolean | string) => {
+            let category = "console";
+            if (error === true) {
+                category = "stderr";
+            }
+            if (typeof error === "string") {
+                category = error;
+            }
+
+            let newLine = "\n";
+            if (category === "stdout" || category === "stderr") {
+                newLine = "";
+            }
+            debugSession.sendEvent(new OutputEvent(message + newLine, category));
+        };
+
+        this.isSettingsInitialized = false;
+    }
+
+    public launch(launchArgs: IDirectLaunchRequestArgs): Promise<void>  {
+        const extProps = {
+            platform: {
+                value: launchArgs.platform,
+                isPii: false,
+            },
+            isDirect: {
+                value: true,
+                isPii: false,
+            },
+            isHermes: {
```

Comment:
Look like this property is redundant here, because there are already platform and isDirect, so when platform === "android" and isDirect === "true" it would mean that Hermes is used.